hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java

```diff
@@ -518,19 +518,19 @@ private static void deleteKeysFromDelDirTableInSnapshotScope(
       String bucketName, BatchOperation batchOperation) throws IOException {
 
     // Range delete start key (inclusive)
-    final String keyPrefix = omMetadataManager.getBucketKeyPrefixFSO(volumeName, bucketName);
-
-    try (TableIterator<String, ? extends Table.KeyValue<String, OmKeyInfo>>
-         iter = omMetadataManager.getDeletedDirTable().iterator(keyPrefix)) {
-      performOperationOnKeys(iter,
-          entry -> {
-            if (LOG.isDebugEnabled()) {
-              LOG.debug("Removing key {} from DeletedDirTable", entry.getKey());
-            }
-            omMetadataManager.getDeletedDirTable().deleteWithBatch(batchOperation, entry.getKey());
-            return null;
-          });
+    final String startKey = omMetadataManager.getBucketKeyPrefixFSO(volumeName, bucketName);
+
+    // Range delete end key (exclusive) - we use the next possible ASCII char after bucket name
+    // Creates the smallest key that is lexicographically larger than the startKey.
+    final String endKey = startKey + Character.MAX_VALUE;
```

Comment:
This is wrong. Next biggest lexicographically largest value would be
We need to do startKey.subString(0, startKey.length()-1) + startKey[lastIndexCharacter] + 1

Reply:
Please add a test case for the same.
Any deleted key starting with startKey + Character.MAX_VALUE should also be deleted